### PR TITLE
pin pip dependencies to specific version and enable dependabot

### PR DESCRIPTION
**Pull request by gdams**
Also removes several unused imports

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,3 @@
-beautifulsoup4
-requests
-cyclonedx-python-lib
-pytest
-black
\ No newline at end of file
+beautifulsoup4==4.12.3
+cyclonedx-python-lib===7.3.1
+Requests==2.31.0
```
Comment (Scanteianu) [inline requirements.txt:3]: should this be capitalized? https://pypi.org/project/requests/ here the install command is lower case

Comment (gdams) [reply requirements.txt:3]: I'll change it, weirdly `pipreqs` generated it like that

Comment (Scanteianu) [reply requirements.txt]: maybe both work

**Review by Scanteianu — APPROVED** [review]
lgtm

**Review by karianna — APPROVED** [review]

**Review by smlambert — APPROVED** [review]
thanks @gdams !

**Pull request merged**